sleap_io/io/nwb.py

```diff
@@ -6,27 +6,298 @@
+
+
+def pose_training_to_labels(pose_training: PoseTraining) -> Labels:  # type: ignore[return]
+    """Creates a Labels object from an NWB PoseTraining object.
+
+    Args:
+        pose_training: An NWB PoseTraining object.
+
+    Returns:
+        A Labels object.
+    """
+    if not isinstance(pose_training, PoseTraining):
+        raise ValueError("The input must be an NWB PoseTraining object.")
+    
+    labeled_frames = []
+    for training_frame in pose_training.training_frames.training_frames.values():
+        video = Video(filename=f"{training_frame.source_video}")
+        frame_idx = training_frame.source_video_frame_index
+        instances = [
+            Instance.from_numpy(
+                points=instance.node_locations,
+                skeleton=nwb_skeleton_to_sleap(instance.skeleton),
+            )
+            for instance in training_frame.skeleton_instances.skeleton_instances.values()
+        ]
+        labeled_frames.append(
+            LabeledFrame(video=video, frame_idx=frame_idx, instances=instances)
+        )
+    return Labels(labeled_frames=labeled_frames)
+
+
+def nwb_skeleton_to_sleap(skeleton: NWBSkeleton) -> SLEAPSkeleton:  # type: ignore[return]
+    """Converts an NWB skeleton to a SLEAP skeleton.
+
+    Args:
+        skeleton: An NWB skeleton.
+
+    Returns:
+        A SLEAP skeleton.
+    """
+    if not isinstance(skeleton, NWBSkeleton):
+        raise ValueError("The input must be an NWB Skeleton object.")
+    
+    nodes = [Node(name=node) for node in skeleton.nodes]
+    edges = [Edge(source=edge[0], destination=edge[1]) for edge in skeleton.edges]
+    return SLEAPSkeleton(
+        nodes=nodes,
+        edges=edges,
+        name=skeleton.name,
+    )
+
+
+def labels_to_pose_training(
+    labels: Labels, img_paths: Optional[list[str]] = None, **kwargs
+) -> PoseTraining:  # type: ignore[return]
+    """Creates an NWB PoseTraining object from a Labels object.
+
+    Args:
+        labels: A Labels object.
+        img_paths: An optional list of image paths for the labeled frames.
+
+    Returns:
+        A PoseTraining object.
+    """
+    if not isinstance(labels, Labels):
+        raise ValueError("The input must be a SLEAP Labels object.")
+    
+    skeletons_list: list[Skeleton] = [] # type: ignore[assignment]
+    skeletons = Skeletons(skeletons=skeletons_list) # type: ignore[assignment]
+    training_frame_list = []
+    for i, labeled_frame in enumerate(labels.labeled_frames):
+        training_frame_name = name_generator("training_frame")
+        training_frame_annotator = f"{training_frame_name}_{i}"
+        skeleton_instances_list = []
+        
+        for instance in labeled_frame.instances:
+            if isinstance(instance, PredictedInstance):
+                continue
+            
+            if instance.skeleton not in skeletons_list:
+                skeletons_list.append(instance.skeleton)
+                skeletons = Skeletons(skeletons=skeletons_list)
```

Comment:
Shouldn't this be outside of the loop? Otherwise `skeletons` gets recreated for every instance.

Also, `instance.skeleton` here will be a `SLEAPSkeleton`, not an `NWBSkeleton`. Use the skeleton container approach from the earlier function to prevent creation of duplicate objects.